### ISSUE TITLE
border color of Textfields not custamizable

### ISSUE THREAD

**Issue by Wissperwind** [flutter/flutter#78870]
If I have a container, I can set the border to a specific color:
```
BoxDecoration blackBorder = BoxDecoration(border: Border.all(color: Colors.black45, width: 0.5),
color: Colors.white, borderRadius: BorderRadius.all(Radius.circular(2)));
```
In Border.all I define that the border color is black45.

As far as I know this parameter is missing for the border of TextFields:
```
 var textFiledBorderDefinition = OutlineInputBorder(borderRadius: BorderRadius.circular(2));
      TextField searchTextField = TextField(
        obscureText: false,
        decoration: InputDecoration(border: textFiledBorderDefinition, fillColor: Colors.white, filled: true, hoverColor: Colors.white, labelText: 'Suche',),
        controller: searchController,
        onSubmitted: onSearchButtonPressed,
      );
```

I can't set a color: in OutlineInputBorder(). So my TextField border is always black. But it need to look exactly as the abve mentioned container border.

<img width="589" alt="Unbenannt 2" src="https://user-images.githubusercontent.com/20759873/112161660-a4007180-8beb-11eb-9385-ecd071cdef2b.PNG">

**Comment by Wissperwind** [comment]
I also haven't found an option to make the border thicker and thinner. Like the width parameter in Border.all().

**Comment by TahaTesser** [comment]
Hi @Wissperwind 
You can change border decoration based on different states such active, disable, focused 

Try this code sample I created for you

<details> 
<summary>code sample</summary> 

```dart
import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Material App',
      home: Home(),
    );
  }
}

class Home extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('Material App Bar'),
      ),
      body: Center(
        child: TextField(
          obscureText: false,
          decoration: InputDecoration(
              enabledBorder: OutlineInputBorder(
                borderRadius: BorderRadius.circular(12),
                borderSide: BorderSide(color: Colors.purple[100], width: 4),
              ),
              focusedBorder: OutlineInputBorder(
                borderRadius: BorderRadius.circular(12),
                borderSide: BorderSide(color: Colors.purple[700], width: 6),
              ),
              disabledBorder: OutlineInputBorder(
                borderRadius: BorderRadius.circular(12),
                borderSide: BorderSide(color: Colors.grey[100], width: 4),
              )),
        ),
      ),
      floatingActionButton: FloatingActionButton(
        child: Icon(Icons.add),
        onPressed: () {},
      ),
    );
  }
}

``` 

</details>

Please see https://flutter.dev/community for resources and asking questions like this,
you may also get some help if you post it on Stack Overflow and if you need help with your code, please see https://www.reddit.com/r/flutterhelp/
Closing, as this isn't an issue with Flutter itself. If you disagree, please write in the comments and I will reopen it.
Thank you

**Comment by Wissperwind** [comment]
Ah, I see. Thanks.

**Comment by Wissperwind** [comment]
But unfortunately I can't improve my visual experience.
I need to get a verry thin border like the designer designed it. On Containers I can set it to width: 0.5 what is round about 1 pixel at a 4k screen. But on TextFields values smaller 1 have the same effect as 1. Is it possible to make tham act liek the containers?

**Comment by TahaTesser** [comment]
@Wissperwind 
Making it that thin doesn't follow material guidelines, you can probably have a custom textfield. You can find help on Stackoverflow or flutterhelp subreddit

**Comment by Wissperwind** [comment]
I have found my problem. I used the border: property. And that wasn't the right one. I had to use enabeledBorder.
I did not understand why there is a general borde property besides the specific one (enabeledBorder , disabeledBorder...)
Maybe that property shoould be removed / renamed.
Material design is cool but I think flutter as any web framework shall be customizable enough to fulfill custom design ideas.
